const.go

```diff
@@ -13,4 +13,8 @@ const (
 
 	// MaxSatoshi is the maximum transaction amount allowed in satoshi.
 	MaxSatoshi = 21e6 * SatoshiPerBitcoin
+
+	// Satoshi is the base bitcoin monetary unit.
+	// A single amount is equal to 1e-8 of a bitcoin.
```

Comment:
@davecgh Renaming it BitcoinPerSatoshi will be consistent with the rest of the code. Makes sense.
@jrick The satoshi constant is a workaround for the quirky behavior of NewAmount(). Conventionally, all amounts are specified int the base unit; Satoshi. Other units are derivative of that. NewAmount() however expects its monetary value specified in BTC. This is why the Satoshi=1e-8 workaround exists. 
Once NewAmount() takes in Satoshi as default, the `const Satosh = 1` will be true and no more needed.